doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -762,58 +683,287 @@ the conversion in hypotheses :n:`{+ @ident}`.
+      ``native_compute_profile.data``. The actual filename used
+      will contain extra characters to avoid overwriting an existing file; that
+      filename is reported to the user.
+      That means you can individually profile multiple uses of
+      :tacn:`native_compute` in a script. From the Linux command line, run ``perf report``
+      on the profile file to see the results. Consult the ``perf`` documentation
+      for more details.
+
+Computing in a term: eval and Eval
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Evaluation of a term can be performed with:
+
+.. tacn:: eval @red_expr in @term
+
+
+   .. insertprodn red_expr red_expr
+
+   .. prodn::
+      red_expr ::= lazy {? @reductions }
+      | cbv {? @reductions }
+      | compute {? @delta_reductions }
+      | vm_compute {? {| @reference_occs | @pattern_occs } }
+      | native_compute {? {| @reference_occs | @pattern_occs } }
+      | red
+      | hnf
+      | simpl {? @delta_reductions } {? {| @reference_occs | @pattern_occs } }
+      | cbn {? @reductions }
+      | unfold {+, @reference_occs }
+      | fold {+ @one_term }
+      | pattern {+, @pattern_occs }
+      | @ident
+
+   :tacn:`eval` is a :token:`value_tactic`.  It returns the result of
+   applying the conversion rules specified by :n:`@red_expr`.  It does not
+   change the proof state.
+
+   The :n:`@red_expr` alternatives that begin with a keyword correspond to the
+   tactic with the same name, though in several cases with simpler syntax
+   than the tactic.  :n:`@ident` is a named reduction expression created
+   with :cmd:`Declare Reduction`.
+
+   .. seealso:: Section :ref:`applyingconversionrules`.
+
+.. cmd:: Eval @red_expr in @term
+
+   Performs the specified reduction on :n:`@term` and displays
+   the resulting term with its type. If a proof is open, :n:`@term`
+   may reference hypotheses of the selected goal.  :cmd:`Eval` is
+   a :token:`query_command`, so it may be prefixed with a goal selector.
+
+   .. cmd:: Compute @term
+
+      Evaluates :n:`@term` using the bytecode-based virtual machine.
+      It is a shortcut for :cmd:`Eval` :n:`vm_compute in @term`.
+      :cmd:`Compute` is a :token:`query_command`, so it may be prefixed
+      with a goal selector.
+
+.. _controlling-the-reduction-strategies:
+
+Controlling reduction strategies and the conversion algorithm
+-------------------------------------------------------------
+
+The commands to fine-tune the reduction strategies and the lazy conversion
+algorithm are described in this section.  Also see :ref:`Args_effect_on_unfolding`,
+which supports additional fine-tuning.
+
+.. cmd:: Opaque {+ @reference }
+
+   Marks the specified constants as :term:`opaque` so tactics won't :term:`unfold` them
+   with :term:`delta-reduction`.
+   "Constants" are items defined by commands such as :cmd:`Definition`,
+   :cmd:`Let` (with an explicit body), :cmd:`Fixpoint`, :cmd:`CoFixpoint`
+   and :cmd:`Function`.
```

Comment:
In the introduction of https://coq.github.io/doc/master/refman/language/core/definitions.html#top-level-definitions, "constant" and "body" are in italic and could be turned into glossary terms. Then we wouldn't have to redefine everywhere what a "constant" is.

Reply:
Don't you think we use "constant" with multiple meanings?  For example, in `changes.rst`, "New tactics ring, ring_simplify and new tactic field now able to manage power to a positive integer **constant**.".

If the definition is "The name defined by the definition is called a
*constant*", then IIUC "2" is not a constant, right?

There are 177 uses of "constant" in the doc; it would take some time to review each of them.  Maybe I could change them en-masse and let you tell me which ones should not point to the defined term?  There are 65 uses of "body", maybe I could mark those, too?

Reply:
If you exclude the changes and history files + the micromega, ring, omega and coq-library files, then I think most uses are consistent (from what I could see with a quick grep).